.github/workflows/pre-commit-config-infra.yaml

```
@@ -0,0 +1,57 @@
+
+on:
+  workflow_call:
+
+jobs:
+  infrastructure-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.8)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)